src/helper_startup.py

```diff
@@ -30,103 +30,98 @@ def _loadTrustedPeer():
 
 
 def loadConfig():
+    config = BMConfigParser()
     if state.appdata:
-        BMConfigParser().read(state.appdata + 'keys.dat')
+        config.read(state.appdata + 'keys.dat')
         # state.appdata must have been specified as a startup option.
-        try:
-            BMConfigParser().get('bitmessagesettings', 'settingsversion')
-            print 'Loading config files from directory specified on startup: ' + state.appdata
-            needToCreateKeysFile = False
-        except Exception:
-            needToCreateKeysFile = True
-
+        needToCreateKeysFile = config.safeGet(
+            'bitmessagesettings', 'settingsversion') is None
+        if not needToCreateKeysFile:
+            print(
+                'Loading config files from directory specified'
+                ' on startup: %s' % state.appdata)
     else:
-        BMConfigParser().read(paths.lookupExeFolder() + 'keys.dat')
+        config.read(paths.lookupExeFolder() + 'keys.dat')
         try:
-            BMConfigParser().get('bitmessagesettings', 'settingsversion')
+            config.get('bitmessagesettings', 'settingsversion')
             print 'Loading config files from same directory as program.'
             needToCreateKeysFile = False
             state.appdata = paths.lookupExeFolder()
-        except Exception:
+        except:
             # Could not load the keys.dat file in the program directory.
             # Perhaps it is in the appdata directory.
             state.appdata = paths.lookupAppdataFolder()
-            BMConfigParser().read(state.appdata + 'keys.dat')
-            try:
-                BMConfigParser().get('bitmessagesettings', 'settingsversion')
+            config.read(state.appdata + 'keys.dat')
+            needToCreateKeysFile = config.safeGet(
+                'bitmessagesettings', 'settingsversion') is None
+            if not needToCreateKeysFile:
                 print 'Loading existing config files from', state.appdata
-                needToCreateKeysFile = False
-            except Exception:
-                needToCreateKeysFile = True
 
     if needToCreateKeysFile:
+
         # This appears to be the first time running the program; there is
         # no config file (or it cannot be accessed). Create config file.
-        BMConfigParser().add_section('bitmessagesettings')
-        BMConfigParser().set('bitmessagesettings', 'settingsversion', '10')
-        BMConfigParser().set('bitmessagesettings', 'port', '8444')
-        BMConfigParser().set(
-            'bitmessagesettings', 'timeformat', '%%c')
-        BMConfigParser().set('bitmessagesettings', 'blackwhitelist', 'black')
-        BMConfigParser().set('bitmessagesettings', 'startonlogon', 'false')
+        config.add_section('bitmessagesettings')
+        config.set('bitmessagesettings', 'settingsversion', '10')
```

Comment:
Why can't we have all this constants some where and loop them ?

Reply:
It's a compatibility layer for previous versions. To turn it into "constants" I would be forced to define a dictionary of dictionaries. It would look awful I think.